tftypes/value.go

```diff
@@ -228,6 +228,31 @@ func (val Value) Equal(o Value) bool {
 	return deepEqual
 }
 
+// SafeEqual returns true if two Values should be considered equal. Values are
+// considered equal if their types are considered equal and if they represent
+// data that is considered equal.
+//
+// This function will never panic.
+func (val Value) SafeEqual(o Value) bool {
```

Comment:
I _feel_ like we should be able to just update the `(tftypes.Value).Equal` function to not panic, rather than introduce a new function....

That is _technically_ a breaking change if anyone downstream is depending on the behavior of a panic occurring if the value types are different when utilizing a DPT..... which seems unlikely 😄 

Obligatory: https://www.hyrumslaw.com/

Reply:
I appreciate you taking the "safer" approach here first!

I would agree with you that we should update the existing `Equal` implementation to not panic as that is a very atypical approach for equality checking methods. From my perspective, we can just `return false` and consumers can dig into the "why" themselves if they really need to in that situation. We should never require consumers to implement `recover` logic to do that.

The intentional 0-based versioning of this Go module leaves us the leeway to make these potentially breaking changes if we feel like the benefits will outweigh any sort of upgrade burden. I personally feel the benefits are worth it here.

Reply:
Awesome, agreed 👍🏻. Updating the PR to reflect this + adding a changelog